waffle_hub/core/hpo/adapter/optuna/optuna_hpo.py

```diff
@@ -0,0 +1,431 @@
+        return self.hpo_dir / OptunaHPO.HPO_RESULT_FILE
+
+    @cached_property
+    def hpo_config_dir(self) -> Path:
+        """HPO Config Directory"""
+        return self.hpo_dir / OptunaHPO.CONFIG_DIR
+
+    @cached_property
+    def hpo_artifacts_dir(self) -> Path:
+        """HPO Artifacts Directory"""
+        return self.hpo_dir / OptunaHPO.HPO_ARTIFACTS_DIR
+
+    @cached_property
+    def storage_name(self) -> str:
+        """HPO Storage Name"""
+        return f"sqlite:///{self.root_dir}/{self.study_name}/{self.study_name}.db"
+
+    # properties
+    @property
+    def study_name(self) -> str:
+        """HPO Study Name"""
+        return self.__study_name
+
+    @study_name.setter
+    @type_validator(str)
+    def study_name(self, v):
+        self.__study_name = v
+
+    @property
+    def root_dir(self) -> Path:
+        """Root Directory"""
+        return self.__root_dir
+
+    @root_dir.setter
+    @type_validator(Path, strict=False)
+    def root_dir(self, v):
+        self.__root_dir = OptunaHPO.parse_root_dir(v)
+        logger.info(f"HPO root directory : {self.root_dir}")
+
+    @property
+    def direction(self) -> str:
+        """HPO direction"""
+        return self.__direction
+
+    @direction.setter
+    @type_validator(str)
+    def direction(self, v):
+        if v is None:
+            warnings.warn("HPO direction is not set. Set to maximize or minimize.")
+        self.__direction = v
+
+    @property
+    def n_trials(self) -> int:
+        """HPO n_trials"""
+        return self.__n_trials
+
+    @n_trials.setter
+    @type_validator(int)
+    def n_trials(self, v):
+        if v is None:
+            warnings.warn("HPO n_trials is not set. Set to 100.")
+        self.__n_trials = v
+
+    @property
+    def search_space(self) -> dict:
+        """HPO search_space"""
+        return self.__search_space
+
+    @search_space.setter
+    def search_space(self, v):
+        if isinstance(v, (str, PurePath)):
+            # chck if it is yaml or json
+            if Path(v).exists():
+                if Path(v).suffix in [".yaml", ".yml"]:
+                    self.__search_space = io.load_yaml(v)
+                elif Path(v).suffix in [".json"]:
+                    self.__search_space = io.load_json(v)
+                else:
+                    raise ValueError(f"search space file should be yaml or json {v}")
+            else:
+                raise ValueError(f"search space file does not exist {v}")
+        elif isinstance(v, dict):
+            self.__search_space = v
+        elif not isinstance(v, dict):
+            raise ValueError(f"search space should be dict or file path {v}")
+
+    @property
+    def metric(self) -> str:
+        """HPO metric"""
+        return self.__metric
+
+    @metric.setter
+    @type_validator(str)
+    def metric(self, v):
+        if v is None:
+            warnings.warn("HPO metric is not set. Set to None.")
+        self.__metric = v
```

Comment:
If it's 'None', what criteria do you use for HPO?

Reply:
Referring to optuna_hpo.py, if the metric is not input when using the hub, an error is emitted, and when using an objective func other than the hub, None is given.
search space is a value that must be entered.
All other values ​​are saved in the default config saved in optuna config.